### PR TITLE
feat(duckdb): add attach_sqlite method

### LINKED ISSUE #5981: feat: Connect to sqlite database from duckdb
### Is your feature request related to a problem?

I would like to be able to use the python API to have a duckdb session attach to a sqlite database. This is separate from reading an individual table, as is currently supported.

This is currently possible to do via SQL:

```python
import ibis

connection = ibis.duckdb.connect()

connection.raw_sql("""
INSTALL sqlite;
LOAD sqlite;
CALL sqlite_attach('EnsDb.Hsapiens.v108.sqlite');
""")


connection.list_tables()
['protein_domain',
 'uniprot',
...
]
```

### Describe the solution you'd like

I would like there to be some python API for this. Potentially something like:

```python
connection = ibis.duckdb.connect()
connection.register("sqlite://EnsDb.Hsapiens.v108.sqlite")
```

### What version of ibis are you running?

'5.1.0'

### What backend(s) are you using, if any?

DuckDB

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gforsyth**
Resolves #5981 

I opted for the `attach_sqlite` method because I imagine someone will request an equivalent `attach_postgres` at some point.

### CODE REVIEW

File: ibis/backends/duckdb/__init__.py [hunk 1]
```diff
@@ -506,6 +507,39 @@ def read_sqlite(self, path: str | Path, table_name: str | None = None) -> ir.Tab
...
+        self, path: str | Path, overwrite: bool = False, all_varchar: bool = False
+    ) -> None:
+        """Attach a SQLite database to the current DuckDB session.
+
+        Parameters
+        ----------
+        path
+            The path to the SQLite database.
+        overwrite
+            Allow re-scan of SQLite database of the same name.
```
Comment (jcrist) [inline ibis/backends/duckdb/__init__.py:520]: Can you expand this comment slightly, it's not 100% clear to me what this option does.

**Review by jcrist — COMMENTED** [review]
Small docstring nit, otherwise LGTM.

File: ibis/backends/duckdb/__init__.py (hunk 1 above)
Comment (gforsyth) [reply ibis/backends/duckdb/__init__.py]: Done

**Comment by jcrist** [comment]
Hmmm, looks like there's some errors with installing the sqlite extension in the nix builds only. Not sure what the issue is there.

**Review by cpcloud — APPROVED** [review]
Number 6️⃣0️⃣0️⃣0️⃣❗

**Pull request merged**